src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -294,7 +294,8 @@ private void VisitNamedTypeWithoutNullability(INamedTypeSymbol symbol)
 
             //visit the enclosing type if the style requires it
             if (Format.TypeQualificationStyle == SymbolDisplayTypeQualificationStyle.NameAndContainingTypes ||
-                Format.TypeQualificationStyle == SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces)
+                Format.TypeQualificationStyle == SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces ||
+                (Format.TypeQualificationStyle == SymbolDisplayTypeQualificationStyle.NameOnly && symbol.IsExtension))
```

Comment:
>(Format.TypeQualificationStyle == SymbolDisplayTypeQualificationStyle.NameOnly && symbol.IsExtension) [](http://example.com/codeflow?start=16&length=101)

I understand that this is probably gives us desired behavior for specific scenarios, but the condition itself is confusing. Especially for someone who doesn't know that we enumerated all possible `SymbolDisplayTypeQualificationStyle` values that are defined at the moment. I think the condition will have more sense and will be more robust to future changes to the `SymbolDisplayTypeQualificationStyle` enum if we simply use ```symbol.IsExtension``` as the condition.